src/cider/log/framework/logback.clj

```diff
@@ -0,0 +1,110 @@
+(ns cider.log.framework.logback
```

Comment:
I wasn't aware of the async log appender. I think it can be done, but reading https://stackoverflow.com/questions/47372092/when-not-to-use-asyncappender-in-logback-by-default I'm not convinced we will gain much for the added complexity and it's only for Logback.

Maybe its worth to mention that all this is opt in. So, if you don't call `cider-log` and attach an appender to a log framework there is no additional work being done. A production system for example, that ships NREPL middleware, will by default not do any extra work. Only if you connect to it and configure any appenders and consumers.

If you add an appender this is what happens on the hot path:
- A log event is converted to a Clojure data structure
- If you defined any filters for an appender (like a log level, or a regex) a filter function is called to decide if the event should be added to the appenders internal atom. If there are no filters defined, the filter function is the `identity` function.
- If the event passes the filter it will be added to an atom

What happens next depends on if there are any consumers. A consumer is for example added for the `*cider-log*` buffer when you search/stream events to the client.

If there are any consumers, then in the hot path a filter function (again `identity` if there are no filters defined) is applied for each consumer. If the event passes the filter function then the event is sent in an NREPL message to the client. My guess is that sending back the event to the client is probably the most expensive operation. At least that's the only IO operation we are doing here. 

I haven't noticed much slowdown when I use this during development (for which I see this as the main use case). But it really depends how much you log.

The only significant slowdown I noticed is on the Emacs side when you insert super long messages into a buffer. Because of that, I decided to add the `cider-log-max-message-length` variable that truncates messages on the Emacs side.

So I would suggest, let's keep this optimization for another day.

Reply:
Right now, yes this would affect the Clojure code emitting log events. My proposed solution for now would be to either configure the log event filters of either the appender (so it does not add too many events to the in memory atom) and/or the filters of the consumer (so it does not send too many events to the client).

You can definitely bring down the thing if you are logging too crazy, but I think for a typical development system it should perform well enough. 

I tested this a bit by emitting random log events in a couple of threads with this snippet:
https://github.com/r0man/cider-nrepl/blob/cider-log/test/clj/cider/nrepl/middleware/log_test.clj#L424

If I start spawning 5 threads each logging 1000 events with a 10ms delay I don't notice any lag when I edit or move around buffers. The `*cider-log*` buffer fills up with events pretty quickly and I believe logview actually can handle it quite well. 

If you start more threads (I tried this now with 10) at some point logview doesn't colorize the events as quickly anymore and you see this in the cider-log buffer.  Moving the cursor still seems to work fine though.

But eventually it will fall apart. But to be honest, with that amount of log events streaming into the cider-log buffer, this isn't the right tool for the job anymore. You should at least start to apply filters to find the interesting events at this point I believe.